lib/overcommit/hook/prepare_commit_msg/replace_branch.rb

```diff
@@ -17,7 +17,7 @@ def run
       Overcommit::Utils.log.debug("Writing #{commit_message_filename} with #{new_template}")
 
       modify_commit_message do |old_contents|
-        "#{new_template}\n#{old_contents}"
+        "#{new_template} #{old_contents}"
```

Comment:
Do you actually want the space? Won't that result in the old contents getting a leading space, or am I misunderstanding the fix? Should we instead check if `new_template` has a newline and add one if it doesn't?

Reply:
I think we would want the space since the text is replaced using the `replacement_text` value. So unless the user consciously adds a space to the end of that value, it will not be included.

Just for reference, using `git version 2.25.0`, old_contents would be as follows (with the preceding newline):

```text

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch bug/705-extra-newline
# Your branch is up to date with 'origin/bug/705-extra-newline'.
#
# Changes to be committed:
#       modified:   lib/overcommit/hook/prepare_commit_msg/replace_branch.rb
#
```